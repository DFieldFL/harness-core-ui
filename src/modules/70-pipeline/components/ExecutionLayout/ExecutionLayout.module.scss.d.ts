/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly main: string
  readonly splitPane1: string
  readonly splitPane2: string
  readonly pane11: string
  readonly pane12: string
  readonly pane21: string
  readonly pane22: string
  readonly floatingView: string
  readonly stepDetails: string
  readonly draggable: string
  readonly toggle: string
  readonly label: string
}
export default styles
