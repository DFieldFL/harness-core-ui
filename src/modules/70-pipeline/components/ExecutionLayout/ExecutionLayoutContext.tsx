import { createContext, useContext } from 'react'
import type { SetStateAction } from 'react'

export enum ExecutionLayoutState {
  NONE = 'NONE',
  RIGHT = 'RIGHT',
  BOTTOM = 'BOTTOM',
  FLOATING = 'FLOATING'
}

export interface ExecutionLayoutContextParams {
  layout: ExecutionLayoutState
  setLayout(action: SetStateAction<ExecutionLayoutState>): void
  primaryPaneSize: number
  setPrimaryPaneSize(action: SetStateAction<number>): void
  teritiaryPaneSize: number
  setTeritiaryPaneSize(action: SetStateAction<number>): void
}

export const ExecutionLayoutContext = createContext<ExecutionLayoutContextParams>({
  layout: ExecutionLayoutState.NONE,
  setLayout: /* istanbul ignore next */ () => void 0,
  primaryPaneSize: 0,
  teritiaryPaneSize: 0,
  setPrimaryPaneSize: /* istanbul ignore next */ () => void 0,
  setTeritiaryPaneSize: /* istanbul ignore next */ () => void 0
})

export function useExecutionLayoutContext(): ExecutionLayoutContextParams {
  return useContext(ExecutionLayoutContext)
}
