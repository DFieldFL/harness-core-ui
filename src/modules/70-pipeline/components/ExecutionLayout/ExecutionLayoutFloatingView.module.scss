.floatingView {
  position: fixed;
  bottom: 0;
  z-index: 1;
  right: 60px;

  :global(#pipeline-step-details-drag) {
    cursor: move;
    padding: 0;

    &:hover {
      box-shadow: none;
    }
  }

  .stepDetails {
    display: flex;
    align-items: center;
    padding: var(--spacing-none) var(--spacing-small);
    background: var(--white);
    box-shadow: 0 2px 6px var(--grey-400) !important;
    height: var(--spacing-xxlarge);
    border-radius: calc(var(--spacing-xxlarge) / 2) !important;
    user-select: none;
  }

  .toggleButton {
    padding: 0 !important;
  }
}
