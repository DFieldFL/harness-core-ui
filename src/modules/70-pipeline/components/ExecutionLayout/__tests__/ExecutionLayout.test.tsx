import React from 'react'
import { render, fireEvent, act } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'

import ExecutionLayout from '../ExecutionLayout'
import { useExecutionLayoutContext, ExecutionLayoutState } from '../ExecutionLayoutContext'

describe('<ExecutionLayout /> tests', () => {
  test('Does not render third child by default', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">Child 2</div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    expect(getByTestId('child-1').innerHTML).toBe('Child 1')
    expect(getByTestId('child-2').innerHTML).toBe('Child 2')
    expect(() => getByTestId('child-3')).toThrow()
    expect(container).toMatchSnapshot()
  })

  test('Showing third child works', () => {
    function Child2(): React.ReactElement {
      const { setStepDetailsVisibility } = useExecutionLayoutContext()

      return (
        <button data-testid="layout" onClick={() => setStepDetailsVisibility(true)}>
          Show Child 3
        </button>
      )
    }
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">
            <Child2 />
          </div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    act(() => {
      fireEvent.click(getByTestId('layout'))
    })

    expect(getByTestId('child-1').innerHTML).toBe('Child 1')
    expect(getByTestId('child-3').innerHTML).toBe('Child 3')
    expect(container).toMatchSnapshot()
  })

  test('"RIGHT" Layout works', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout defaultLayout={ExecutionLayoutState.RIGHT} defaultStepVisibility>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">Child 2</div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    expect(getByTestId('child-1').innerHTML).toBe('Child 1')
    expect(getByTestId('child-2').innerHTML).toBe('Child 2')
    expect(getByTestId('child-3').innerHTML).toBe('Child 3')
    expect(container.querySelector('.splitPane2')?.classList.contains('vertical')).toBe(true)
    expect(container).toMatchSnapshot()
  })

  // eslint-disable-next-line jest/no-disabled-tests
  test.skip('"BOTTOM" Layout works', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout defaultLayout={ExecutionLayoutState.BOTTOM} defaultStepVisibility>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">Child 2</div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    expect(getByTestId('child-1').innerHTML).toBe('Child 1')
    expect(getByTestId('child-2').innerHTML).toBe('Child 2')
    expect(getByTestId('child-3').innerHTML).toBe('Child 3')
    expect(container.querySelector('.splitPane2')?.classList.contains('horizontal')).toBe(true)
    expect(container).toMatchSnapshot()
  })

  test('"FLOATING" Layout works', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout defaultLayout={ExecutionLayoutState.FLOATING} defaultStepVisibility>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">Child 2</div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    expect(getByTestId('child-1').innerHTML).toBe('Child 1')
    expect(getByTestId('child-2').innerHTML).toBe('Child 2')
    expect(getByTestId('child-3').innerHTML).toBe('Child 3')
    expect(getByTestId('child-3').parentElement?.classList.contains('floating-wrapper')).toBe(true)
    expect(container).toMatchSnapshot()
  })

  test('"FLOATING" Layout - toggle window works', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout defaultLayout={ExecutionLayoutState.FLOATING} defaultStepVisibility>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">Child 2</div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    expect(getByTestId('child-3').innerHTML).toBe('Child 3')
    expect(container).toMatchSnapshot('Child 3 is shown')

    const btn = container.querySelector('button.toggleButton')!

    act(() => {
      fireEvent.click(btn)
    })

    expect(() => getByTestId('child-3')).toThrow()
    expect(container).toMatchSnapshot('Child 3 is not shown')
  })

  test('ExecutionLayout.Toggle works', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout defaultLayout={ExecutionLayoutState.RIGHT} defaultStepVisibility>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">
            <ExecutionLayout.Toggle />
          </div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()

    const [right, /*bottom,*/ floating] = container.querySelectorAll<HTMLInputElement>('input[name="layout"]')

    expect(right.checked).toBe(true)
    expect(container.querySelector('.splitPane2')?.classList.contains('vertical')).toBe(true)

    // act(() => {
    //   fireEvent.click(bottom)
    // })
    // expect(container.querySelector('.splitPane2')?.classList.contains('horizontal')).toBe(true)

    act(() => {
      fireEvent.click(floating)
    })
    expect(getByTestId('child-3').parentElement?.classList.contains('floating-wrapper')).toBe(true)
  })

  test('Minimize works', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <ExecutionLayout defaultLayout={ExecutionLayoutState.RIGHT} defaultStepVisibility>
          <div data-testid="child-1">Child 1</div>
          <div data-testid="child-2">
            <ExecutionLayout.Toggle />
          </div>
          <div data-testid="child-3">Child 3</div>
        </ExecutionLayout>
      </TestWrapper>
    )

    const min = container.querySelector('input[value="MINIMIZE"]')!

    act(() => {
      fireEvent.click(min)
    })

    expect(() => getByTestId('child-3')).toThrow()

    const restore1 = getByTestId('restore')

    act(() => {
      fireEvent.click(restore1)
    })

    expect(getByTestId('child-3')).toBeTruthy()
    // expect(container.querySelector('.splitPane2')?.classList.contains('vertical')).toBe(true)
  })
})
