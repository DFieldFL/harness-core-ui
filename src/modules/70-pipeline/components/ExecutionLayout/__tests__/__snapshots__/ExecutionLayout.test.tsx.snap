// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExecutionLayout /> tests "BOTTOM" Layout works 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            class="SplitPane splitPane2 horizontal "
            style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
          >
            <div
              class="Pane horizontal Pane1  "
              style="flex: 0 0 auto; position: relative; outline: none; height: 500px; display: flex;"
            >
              <div
                class="Pane  pane21"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-2"
                >
                  Child 2
                </div>
              </div>
            </div>
            <span
              class="Resizer horizontal "
              role="presentation"
            />
            <div
              class="Pane horizontal Pane2  "
              style="flex: 1; position: relative; outline: none;"
            >
              <div
                class="Pane  pane22"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-3"
                >
                  Child 3
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionLayout /> tests "FLOATING" Layout - toggle window works: Child 3 is not shown 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-2"
          >
            Child 2
          </div>
          <div
            class="floatingView"
            data-state="floating"
          >
            <div
              class="stepDetails react-draggable"
              style="transform: translate(-40px,-30px);"
            >
              <button
                class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow"
                id="pipeline-step-details-drag"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-drag-handle-vertical StyledProps--main"
                  icon="drag-handle-vertical"
                >
                  <svg
                    data-icon="drag-handle-vertical"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      drag-handle-vertical
                    </desc>
                    <path
                      d="M6 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4-6c.55 0 1-.45 1-1s-.45-1-1-1-1 .45-1 1 .45 1 1 1zM6 13c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0 8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="bp3-button bp3-disabled bp3-loading bp3-minimal bp3-small Button--button StyledProps--font StyledProps--main toggleButton StyledProps--intent-primary Button--with-current-color Button--without-shadow"
                data-testid="restore"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <div
                  class="bp3-spinner bp3-button-spinner"
                >
                  <div
                    class="bp3-spinner-animation"
                  >
                    <svg
                      height="20"
                      stroke-width="16.00"
                      viewBox="-3.00 -3.00 106.00 106.00"
                      width="20"
                    >
                      <path
                        class="bp3-spinner-track"
                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                      />
                      <path
                        class="bp3-spinner-head"
                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                        pathLength="280"
                        stroke-dasharray="280 280"
                        stroke-dashoffset="210"
                      />
                    </svg>
                  </div>
                </div>
                <span
                  class="bp3-button-text"
                >
                  <span>
                    pipeline.stepDetails
                  </span>
                </span>
                <span
                  class="bp3-icon bp3-icon-plus StyledProps--main"
                  icon="plus"
                >
                  <svg
                    data-icon="plus"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      plus
                    </desc>
                    <path
                      d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionLayout /> tests "FLOATING" Layout - toggle window works: Child 3 is shown 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-2"
          >
            Child 2
          </div>
          <div
            class="floatingView"
            data-state="floating"
          >
            <div
              class="stepDetails react-draggable"
              style="transform: translate(-40px,-30px);"
            >
              <button
                class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow"
                id="pipeline-step-details-drag"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-drag-handle-vertical StyledProps--main"
                  icon="drag-handle-vertical"
                >
                  <svg
                    data-icon="drag-handle-vertical"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      drag-handle-vertical
                    </desc>
                    <path
                      d="M6 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4-6c.55 0 1-.45 1-1s-.45-1-1-1-1 .45-1 1 .45 1 1 1zM6 13c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0 8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="bp3-button bp3-minimal bp3-small Button--button StyledProps--font StyledProps--main toggleButton StyledProps--intent-primary Button--with-current-color Button--without-shadow"
                data-testid="restore"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  <span>
                    pipeline.stepDetails
                  </span>
                </span>
                <span
                  class="bp3-icon bp3-icon-minus StyledProps--main"
                  icon="minus"
                >
                  <svg
                    data-icon="minus"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      minus
                    </desc>
                    <path
                      d="M13 7H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="floating-wrapper"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <div
                data-testid="child-3"
              >
                Child 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionLayout /> tests "FLOATING" Layout works 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-2"
          >
            Child 2
          </div>
          <div
            class="floatingView"
            data-state="floating"
          >
            <div
              class="stepDetails react-draggable"
              style="transform: translate(-40px,-30px);"
            >
              <button
                class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow"
                id="pipeline-step-details-drag"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-drag-handle-vertical StyledProps--main"
                  icon="drag-handle-vertical"
                >
                  <svg
                    data-icon="drag-handle-vertical"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      drag-handle-vertical
                    </desc>
                    <path
                      d="M6 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4-6c.55 0 1-.45 1-1s-.45-1-1-1-1 .45-1 1 .45 1 1 1zM6 13c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0 8c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="bp3-button bp3-minimal bp3-small Button--button StyledProps--font StyledProps--main toggleButton StyledProps--intent-primary Button--with-current-color Button--without-shadow"
                data-testid="restore"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  <span>
                    pipeline.stepDetails
                  </span>
                </span>
                <span
                  class="bp3-icon bp3-icon-minus StyledProps--main"
                  icon="minus"
                >
                  <svg
                    data-icon="minus"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      minus
                    </desc>
                    <path
                      d="M13 7H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="floating-wrapper"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <div
                data-testid="child-3"
              >
                Child 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionLayout /> tests "RIGHT" Layout works 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            class="SplitPane splitPane2 vertical "
            style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; flex-direction: row; left: 0px; right: 0px;"
          >
            <div
              class="Pane vertical Pane1  "
              style="flex: 1; position: relative; outline: none;"
            >
              <div
                class="Pane  pane21"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-2"
                >
                  Child 2
                </div>
              </div>
            </div>
            <span
              class="Resizer vertical "
              role="presentation"
            />
            <div
              class="Pane vertical Pane2  "
              style="flex: 0 0 auto; position: relative; outline: none; width: 570px;"
            >
              <div
                class="Pane  pane22"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-3"
                >
                  Child 3
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionLayout /> tests Does not render third child by default 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-2"
          >
            Child 2
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionLayout /> tests ExecutionLayout.Toggle works 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            class="SplitPane splitPane2 vertical "
            style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; flex-direction: row; left: 0px; right: 0px;"
          >
            <div
              class="Pane vertical Pane1  "
              style="flex: 1; position: relative; outline: none;"
            >
              <div
                class="Pane  pane21"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-2"
                >
                  <div
                    class="toggle"
                  >
                    <label
                      class="label"
                      title="pipeline.execution.layouts.right"
                    >
                      <input
                        checked=""
                        name="layout"
                        type="radio"
                        value="RIGHT"
                      />
                      <span
                        class="bp3-icon StyledProps--main"
                        data-icon="layout-right"
                      >
                        <svg
                          height="16"
                          viewBox="0 0 4.233 4.233"
                          width="16"
                        >
                          <g
                            transform="translate(0 .423) scale(.21167)"
                          >
                            <rect
                              fill="#f2f2f2"
                              height="15"
                              rx="0.5"
                              stroke="#9ea9b4"
                              width="19"
                              x="0.5"
                              y="0.5"
                            />
                            <path
                              d="M12 1h7v14h-7z"
                              fill="#9ea9b4"
                            />
                          </g>
                        </svg>
                      </span>
                    </label>
                    <label
                      class="label"
                      title="pipeline.execution.layouts.float"
                    >
                      <input
                        name="layout"
                        type="radio"
                        value="FLOATING"
                      />
                      <span
                        class="bp3-icon StyledProps--main"
                        data-icon="layout-float"
                      >
                        <svg
                          height="16"
                          viewBox="0 0 4.233 4.233"
                          width="16"
                        >
                          <g
                            fill="none"
                            transform="translate(0 .403) scale(.20982)"
                          >
                            <rect
                              fill="#000"
                              fill-opacity="0.05"
                              height="16.333"
                              rx="1"
                              width="20.176"
                            />
                            <rect
                              height="15.333"
                              rx="0.5"
                              stroke="#66798b"
                              stroke-opacity="0.6"
                              width="19.176"
                              x="0.5"
                              y="0.5"
                            />
                            <path
                              d="M9.303 3a1 1 0 011-1h7a1 1 0 011 1v9a1 1 0 01-1 1h-7a1 1 0 01-1-1zM14.303 14.5a.5.5 0 01.5-.5h3a.5.5 0 110 1h-3a.5.5 0 01-.5-.5z"
                              fill="#9ea9b4"
                            />
                          </g>
                        </svg>
                      </span>
                    </label>
                    <label
                      class="label"
                      title="pipeline.execution.layouts.minimize"
                    >
                      <input
                        name="layout"
                        type="radio"
                        value="MINIMIZE"
                      />
                      <span
                        class="bp3-icon bp3-icon-minus StyledProps--main"
                        icon="minus"
                      >
                        <svg
                          data-icon="minus"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            minus
                          </desc>
                          <path
                            d="M13 7H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="Resizer vertical "
              role="presentation"
            />
            <div
              class="Pane vertical Pane2  "
              style="flex: 0 0 auto; position: relative; outline: none; width: 570px;"
            >
              <div
                class="Pane  pane22"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-3"
                >
                  Child 3
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionLayout /> tests Showing third child works 1`] = `
<div>
  <div
    class="main"
    id="execution-layout-test"
  >
    <div
      class="SplitPane splitPane1 horizontal "
      style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
    >
      <div
        class="Pane horizontal Pane1  "
        style="flex: 0 0 auto; position: relative; outline: none; height: 250px; display: flex;"
      >
        <div
          class="Pane  pane11"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            data-testid="child-1"
          >
            Child 1
          </div>
        </div>
      </div>
      <span
        class="Resizer horizontal "
        role="presentation"
      />
      <div
        class="Pane horizontal Pane2  "
        style="flex: 1; position: relative; outline: none;"
      >
        <div
          class="Pane  pane12"
          style="flex: 1; position: relative; outline: none;"
        >
          <div
            class="SplitPane splitPane2 vertical "
            style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: hidden; user-select: text; flex-direction: row; left: 0px; right: 0px;"
          >
            <div
              class="Pane vertical Pane1  "
              style="flex: 1; position: relative; outline: none;"
            >
              <div
                class="Pane  pane21"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-2"
                >
                  <button
                    data-testid="layout"
                  >
                    Show Child 3
                  </button>
                </div>
              </div>
            </div>
            <span
              class="Resizer vertical "
              role="presentation"
            />
            <div
              class="Pane vertical Pane2  "
              style="flex: 0 0 auto; position: relative; outline: none; width: 570px;"
            >
              <div
                class="Pane  pane22"
                style="flex: 1; position: relative; outline: none;"
              >
                <div
                  data-testid="child-3"
                >
                  Child 3
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
